app/models/subject.rb

```diff
@@ -22,8 +22,7 @@ class Subject < ApplicationRecord
   belongs_to :parent_subject, class_name: "Subject", optional: true
 
   has_many :child_subjects, class_name: "Subject", foreign_key: :parent_subject_id, dependent: :destroy
-  has_many :placement_subject_joins, dependent: :restrict_with_exception
-  has_many :placements, through: :placement_subject_joins # TODO: Remove `through: :placement_subject_joins` after data migration
+  has_many :placements
```

Comment:
Any reason for dropping the `dependent: :restrict_with_exception`?

It could be handy just to make sure we never accidentally delete a subject – for example, when using the Rails console.